spyder/plugins/explorer/widgets.py

```diff
@@ -155,30 +156,71 @@ def icon(self, icontype_or_qfileinfo):
 
                 if extension in self.OFFICE_FILES:
                     icon = ima.icon(self.OFFICE_FILES[extension])
-
-                if mime_type is not None:
-                    try:
-                        # Fix for issue 5080.  Even though mimetypes.guess_type
-                        # documentation states that the return value will be
-                        # None or a tuple of the form type/subtype, in the
-                        # Windows registry, .sql has a mimetype of text\plain
-                        # instead of text/plain therefore mimetypes is
-                        # returning it incorrectly.
-                        file_type, bin_name = mime_type.split('/')
-                    except ValueError:
-                        file_type = 'text'
-                    if file_type == 'text':
-                        icon = ima.icon('TextFileIcon')
-                    elif file_type == 'audio':
-                        icon = ima.icon('AudioFileIcon')
-                    elif file_type == 'video':
-                        icon = ima.icon('VideoFileIcon')
-                    elif file_type == 'image':
-                        icon = ima.icon('ImageFileIcon')
-                    elif file_type == 'application':
-                        if bin_name in self.application_icons:
-                            icon = ima.icon(self.application_icons[bin_name])
-                return icon
+                        
+                language = { 
+                    '.c': 'CFileIcon',
+                    '.cc': 'CFileIcon',
+                    '.cpp': 'CppFileIcon',
+                    '.cxx': 'CppFileIcon',

```

Comment:
We need to expand the list of extensions associated with `CppFileIcon`. Please see

https://en.wikibooks.org/wiki/C%2B%2B_Programming/Programming_Languages/C%2B%2B/Code/File_Organization#Extensions

for the most common used ones.

@dalthviz, I think to simplify the implementation, we could use a dict of dicts here instead of a single dict. What do you think?

Reply:
I was thinking that we could group file extensions in a dict and associate that with its corresponding icon, like this:

```python
{{'.c', '.h'}: 'CFileIcon'}
```

but I forgot dictionaries are not hashable, sorry.

For now please move this dictionary to `icon_manager` and call it `LANGUAGE_ICONS` because we're going to need it for the file switcher too.